dbms/src/Common/HashTable/HashTable.h

```diff
@@ -851,6 +854,17 @@ class HashTable
 
     iterator end() { return iterator(this, buf ? buf + grower.bufSize() : buf); }
 
+    void ALWAYS_INLINE prefetch(size_t hashval) const
+    {
+        (void)hashval;
+#if defined(_MSC_VER) && (defined(_M_X64) || defined(_M_IX86))
+        size_t place_value = grower.place(hashval);
+        __mm_prefetch((const char *)(&buf[place_value]), _MM_HINT_NTA);
```

Comment:
Why use `_MM_HINT_NTA`? My tests found that `_MM_HINT_T0` is usually better than `_MM_HINT_NTA` even when the memory accesses are totally random.
Besides, I think using `__builtin_prefetch` is enough for GCC/Clang.